xclim/indices/_threshold.py

```diff
@@ -395,10 +493,87 @@ def compute_gsl(yrdata):
         )  # If gs is not ended at end of year
         return sl.where(sl >= 0)  # When end happens before beg.
 
-    gsl = tas.resample(time=freq).map(compute_gsl)
+    gsl = tas.resample(time=freq).map(compute_growing_season_length)
     return gsl
 
 
+@declare_units("days", tas="[temperature]", thresh="[temperature]")
+def date_of_last_spring_frost(
+    tas: xarray.DataArray,
+    thresh: str = "0 degC",
+    mid_date: str = "07-01",
+    window: int = 1,
+    freq: str = "YS",
+):
+    r"""Last day of temperatures inferior to a threshold temperature.
+
+    Returns last day of period where a temperature is inferior to a threshold
+    over a given number of days and limited to a final calendar date.
+
+    Parameters
+    ----------
+    tas : xarray.DataArray
+      Mean daily temperature [℃] or [K].
+    thresh : str
+      Threshold temperature on which to base evaluation [℃] or [K]. Default '0 degC'.
+    mid_date : str
```

Comment:
How will this work for global data sets ?

Reply:
Presently this will only work in the Northern hemisphere. I'll make a note of this.

Reply:
Another option is to set the mid_date based on the latitude coordinate.

Reply:
In my implementation of `growing_season_length`, I added a warning in the doc and an example for the southern hemisphere. We could get the latitude coord automatically, but in the case of a global dataset, that would mean a discontinuity in the indicator algorithm at 0°.  I prefer letting the user manage this.

Reply:
So it would be interesting to have some feedback from colleagues about this.

Reply:
The behaviour of this function is now a lot more consistent, but this issue still remains. Can we tag this as resolved once I open an issue about this particular problem? (Dealing with this is well beyond the scope of this PR)